### PR TITLE
Feature/setup

### PR DESCRIPTION
**Pull request by Aquinasmoriasi**
- Added models, views and controllers for User, Groups and Expenses
- Added joining table for groups and expenses table
- Added styling to the application
- Secured app form n+1 problems
- Added specs for the views, models and controllers
- This is the [Video Description](https://drive.google.com/file/d/1F4MBKh3n99uzd6452uHiHvoDeEz7OcFd/view?usp=share_link)

0937B8AC-D3A2-4104-98A0-AA079C4EF10F
47EEA2B7-757B-4E17-A0F2-1AD300E5AE57
AF4D8BB4-AC9A-4FDF-830B-D977536D745C
A5BB080D-E943-40CC-9D77-9045B006F409
24DD9E8D-51BC-4C24-8754-42C1BF06A99C
ACFD6ED8-CBFF-4161-896D-73A84E1F0EEB

### CODE REVIEW

File: spec/views/groups/groups_spec.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+RSpec.describe 'Groups', type: :feature do
+  describe 'GET groups#index' do
+    before(:each) do
+      user = User.create name: 'anonymus', email: 'anonymus@anon.co', password: 'password'
+
+      visit new_user_session_path
+      page.fill_in placeholder: 'Email', with: user.email
+      page.fill_in placeholder: 'password', with: user.password
+      click_button 'Save'
```
Comment (rloterh) [inline spec/views/groups/groups_spec.rb]: - [ ] Two of the view specs happen to be failing. Kindly ensure that **all** test cases are passing.

**Failing test cases:**
image

File: README.md [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+```
+rspec <relative path to the test file>
+```
+
+### Deployment
+
+This project is deployed at [https://stormy-plateau-90686.herokuapp.com/](https://stormy-plateau-90686.herokuapp.com/)
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
```
Comment (rloterh) [inline README.md:142]: - [ ] Kindly ensure to add the _Future Features_ section to your ReadMe, which should include 1-3 potential features to be added

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Moriasi, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project, but you are almost there! 🚦

## Highlights 👍  
- Good presentation of project
- Good job acknowledging [Gregoire Vella](https://www.behance.net/gregoirevella) in your ReadMe
- Properly added required models, controllers, and corresponding views
- Devise correctly implemented for authentication
- Schema follow given ERD
- Nice job implementing solutions to `N + 1` problem
- Design matches template design
- App is running as intended
- Correctly deployed app
- Good ReadMe

## Required Changes ♻️
_Kindly check the inline comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DeliceLydia — APPROVED** [review]
## Hi @Aquinasmoriasi, 

In my opinion, your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


### Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Comment by Aquinasmoriasi** [comment]
Thank you.

**Pull request merged**